### PR TITLE
Pipegun updates and glowup

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Updates pipeguns with a brand new set of sprites. Also comes with pipe pistols; plinkier versions of the pipegun.

ye pipeguns

image

image

Alters pipeguns away from utilizing shotgun/rifle ammunition towards a bespoke junk round ammo. These shells are crafted similarly to current improvised shells. This also removes improvised shells as a shotgun ammo type. (these new shells aren't shotgun shells to begin with, they're more like rifle bullets)

Crafting junk rounds produces an entirely unknown projectile in the shell. You won't know what you get until you fire the gun. Almost in every instance, however, the shot is beneficial towards killing things. And they all do about the same amount of damage, so there is hopefully never a round you didn't want to fire in the gun. Fighting someone with a pipegun is usually predictable for what amount of damage it will do, but any additional effects it might have is going to be an unknown factor. They also brutalize borgs, which is a quality that improvised shells had previously.

Pipeguns operate as they did before, and do roughly 40 damage per shot with the majority of their ammo types (less than what they do currently with .310). They only have one shell in the gun at a time, so every time they're fired, they must be reloaded.

Pipe pistols do roughly 15 damage per shot, but hold more ammo than the bigger pipegun. However, they're more likely to veer off-course.

There are regal versions of each of these weapons, and each are more potent respective of their general rarity. Having one is going to be a lot more noteworthy. 

### Minor changes

Some more clothing items can carry pipeguns in their suit storage.

The icemoon hermit spawns with a heroic laser musket rather than a regal pipegun (partially because it is too difficult for the hermit to get more ammo for their gun, and also because the regal pipegun is a more powerful weapon than previously)

## Why It's Good For The Game

It has been a few years since I added the pipegun, and time was never particularly kind to it. There are alternative weapons now with interesting mechanics of their own. The ammunition it used has changed considerably. And it simply didn't ever feel like a 'junk gun' in a way that was fun. The original mechanics added to give it that feeling were just not fun to experience and were removed. So the pipegun has been left as 'a shitty version of X'.  Even the regal pipegun was, at the end of the day, a shittier version of either a shotgun or cargo rifle.

It didn't feel right not having some kind of unique quality to using these weapons that help them express themselves as unpredictable trash weapons built inside of maintenance. But I expressly didn't want to make it 'unpredictable' in a way that felt unfair on the person getting shot either. So just giving it more damage was right out. As a compromise, I reduced the overall lethality of the weapons while introducing a gimmick that will appeal to those wanting to play out the role of a homeless lunatic with a shitty homemade gun stalking maintenance. A 'chaos gun' so to speak. Maybe that will appeal to people.

## Changelog
:cl:
balance: Reworks pipeguns to use an unpredictable 'junk round'. You won't know what you're shooting until you fire it.
add: Introduces a pistol version of the pipegun; the pipe pistol. It is inaccurate and does significantly less damage, but more portable and has more ammunition in the gun.
image: Updates the visuals of the pipeguns.
balance: Also improves the Regal varieties of these weapons. By a lot.
balance: More articles of clothing can be used to carry pipeguns in suit storage.
balance: The Icemoon Hermit comes with a Heroic Laser Musket instead of a Regal Pipegun.
remove: Improvised shells (the shotgun shell) has been replaced with improvised junk shells (which don't work with shotguns but do work with pipeguns).
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
I've literally no clue as to why CI is yelling at me right now, but if I just take out the recipe for the regal pipe pistol and this suddenly starts to compile, I'll literally just add it at a later date and not give a single fuck.

**Comment by SimplySmiley123** [comment]
So this is indirectly killing lethal shotguns even more?
Or am I missing something.

**Comment by SomeoneYouProbablyKnow** [comment]
> So this is indirectly killing lethal shotguns even more?

Good

File: code/modules/cargo/bounties/assistant.dm [hunk 1]
```diff
@@ -198,11 +198,11 @@
 	wanted_types = list(/obj/item/pneumatic_cannon/ghetto = TRUE)
 
 /datum/bounty/item/assistant/improvised_shells
-	name = "Improvised Shotgun Shells"
-	description = "Budget cuts are hitting our security department pretty hard. Send some improvised shotgun shells when you can."
+	name = "Junk Shells"
+	description = "Our assisstant militia has chewed through all our iron supplies. To stop them making bullets out of station property, we need junk shells, pronto."
```
Comment (Jacquerel) [inline code/modules/cargo/bounties/assistant.dm:202]:
```suggestion
	description = "Our assistant militia has chewed through all our iron supplies. To stop them making bullets out of station property, we need junk shells, pronto."
```

File: code/modules/projectiles/guns/ballistic/rifle.dm [hunk 2]
```diff
@@ -259,46 +259,78 @@
...
-	desc = "An excellent weapon for flushing out tunnel rats and enemy assistants, but its rifling leaves much to be desired."
-	icon = 'icons/obj/weapons/guns/ballistic.dmi'
-	icon_state = "musket"
-	inhand_icon_state = "musket"
-	worn_icon_state = "musket"
-	lefthand_file = 'icons/mob/inhands/weapons/64x_guns_left.dmi'
-	righthand_file = 'icons/mob/inhands/weapons/64x_guns_right.dmi'
-	inhand_x_dimension = 64
-	inhand_y_dimension = 64
+	desc = "A symbol that the true masters of this place are not those who merely inhabits it. But the one willing to twist it towards a killing intent."
```
Comment (Jacquerel) [inline code/modules/projectiles/guns/ballistic/rifle.dm:266]:
```suggestion
	desc = "A symbol that the true masters of this place are not those who merely inhabit it, but the one willing to twist it towards a killing intent."
```

File: code/modules/projectiles/projectile/bullets/junk.dm [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+/obj/projectile/bullet/junk
+	name = "junk bullet"
+	icon_state = "trashball"
+	damage = 30
+	embedding = list(embed_chance=15, fall_chance=3, jostle_chance=4, ignore_throwspeed_threshold=TRUE, pain_stam_pct=0.4, pain_mult=5, jostle_pain_mult=6, rip_time=10)
+	var/bane_multiplier = 1.5
+	var/bane_added_damage = 0
+
+/obj/projectile/bullet/junk/shock/Initialize(mapload)
```
Comment (Jacquerel) [inline code/modules/projectiles/projectile/bullets/junk.dm:11]: Can you move this next to its type declaration?

File: code/modules/projectiles/projectile/bullets/junk.dm [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	if(isliving(target))
+		var/mob/living/victim = target
+		victim.electrocute_act(damage, src, siemens_coeff = 1, flags = SHOCK_NOSTUN)
+
+/obj/projectile/bullet/junk/hunter
+	name = "junk hunter bullet"
+
+/obj/projectile/bullet/junk/hunter/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/bane, mob_biotypes = MOB_BEAST, target_type = /mob/living, damage_multiplier = 0, added_damage = 50)
```
Comment (Jacquerel) [inline code/modules/projectiles/projectile/bullets/junk.dm:44]: Why doesn't this use the bane vars you put on the basetype?
Alternately, is it worth having those on the basetype if only two children use it?

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Mostly grammar

**Comment by carlarctg** [comment]
Need me a flintlock baldric belt so I can store 4 pistols and cycle between them without reloading

**Comment by OuroborusReporter** [comment]
Finally, now when I run out of ammo while fighting ops, I can pull my smoothbore junk revolver and miss entirely, nailing the mime

File: code/modules/projectiles/projectile/bullets/junk.dm (hunk 4 above)
Comment (necromanceranne) [reply code/modules/projectiles/projectile/bullets/junk.dm:44]: Oh I forgot to do that, shit.

File: code/modules/projectiles/projectile/bullets/junk.dm (hunk 3 above)
Comment (necromanceranne) [reply code/modules/projectiles/projectile/bullets/junk.dm:11]: I'm daft as fuck this is meant to be on all subtypes. I'll fix

**Comment by Loafin34** [comment]
Does the junk shot have the same recipe as the improvised scatter shell? Also, please make a broken pipe we can affix on our rifles as a bayonet. It would be funny!

**Comment by necromanceranne** [comment]
yep, same recipe

**Comment by newfren** [comment]
what effects can junk round get?

**Comment by necromanceranne** [comment]
Oh shit, I hadn't actually listed those just yet, sorry. 

- Basic Junk Round (Common): It's the standard round. Most every junk round has Robot bane on it, dealing x1.5 extra damage to robots.
- Incendiary Junk Round (Uncommon): It shoots a big gout of fire, pretty self explanatory honestly.
- Shock Junk Round (Uncommon): It electrocutes the target. This isn't a stun, it's just mixed brute and burn damage. And I guess anyone with the Insulated mutation is immune to that extra damage. Whatever.
- Hunter Junk Round (Uncommon): A normal junk round but with Beast and Special Mob bane. This makes them very good at killing fauna, megafauna ~~and a few other mobs but I doubt it will come up much because this isn't reliably obtainable~~.
- Phasic Junk Round (Rare): Goes through walls, much like x-ray lasers do. I really doubt this will play into anything very much since you won't know you even have a phasic round, but you might hit someone in a room over on accident if you miss and that's fucking funny.
- Ripper Junk Round (Rare): It does less damage but opens horrific wounds and almost always embeds.
-  Reaper Junk Round (Once every couple hundred something rounds maybe): Hitscan bullet, similar to the 10mm Reaper rounds you get in Regal Condors. Doesn't do any extra damage, it's just hitscan. Otherwise a normal junk bullet.

Bane is broken right now so uh hopefully by the time this is merged that's fixed, iunno, sorries.

**Comment by EEASAS** [comment]
i like it

**Comment by projectkepler-ru** [comment]
Genuinely a great idea, why didn't we have pipe pistol? plus, this is really slick sprite













I use pipegun for self defense since that's what the founding maintainer intended
four assistants break into engineering
what the devil
as i grab my welding hardhat and pipe rifle
Blow a golf ball size hole thrugh the first man, he received weeping avulsion and bled out on the spot
draw my pistol on the second man, miss him entirely and hit a dogborg 
I had to resort to the cannon mounted by the door, tally ho lads!
the sound and shrapnel set off the firealarm
I charge at the last man with my plasma cutter and delimbed him in two hit, he become a nugget waiting for seccies to arrive because 25% delimb chance is impossible to have any balancing impact

**Review by Wallemations — APPROVED** [review]

**Pull request merged**